lib/address.ex

```diff
@@ -66,4 +86,13 @@ defmodule Bitcoinex.Address do
         false
     end
   end
+
+  @spec decode_type(String.t(), Bitcoinex.Network.network_name()) ::
+          {:ok, address_type} | {:error, term()}
```

Comment:
the only possible error is `:decode_error`. Let's change `term()` to `:decode_error` that help people using dialyzer for type checking.